x/slashing/keeper_test.go

```diff
@@ -0,0 +1,111 @@
+package slashing
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	abci "github.com/tendermint/abci/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/stake"
+)
+
+func TestHandleDoubleSign(t *testing.T) {
+	ctx, ck, sk, keeper := createTestInput(t)
+	addr, val, amt := addrs[0], pks[0], int64(100)
+	got := stake.NewHandler(sk)(ctx, newTestMsgDeclareCandidacy(addr, val, amt))
+	require.True(t, got.IsOK())
+	_ = sk.Tick(ctx)
+	require.Equal(t, ck.GetCoins(ctx, addr), sdk.Coins{{sk.GetParams(ctx).BondDenom, initCoins - amt}})
+	require.Equal(t, sdk.NewRat(amt), sk.Validator(ctx, addr).GetPower())
+	keeper.handleDoubleSign(ctx, 0, 0, val) // double sign less than max age
+	require.Equal(t, sdk.NewRat(amt).Mul(sdk.NewRat(19).Quo(sdk.NewRat(20))), sk.Validator(ctx, addr).GetPower())
+	ctx = ctx.WithBlockHeader(abci.Header{Time: 300})
+	keeper.handleDoubleSign(ctx, 0, 0, val) // double sign past max age
+	require.Equal(t, sdk.NewRat(amt).Mul(sdk.NewRat(19).Quo(sdk.NewRat(20))), sk.Validator(ctx, addr).GetPower())
+}
+
+func TestHandleAbsentValidator(t *testing.T) {
+	ctx, ck, sk, keeper := createTestInput(t)
+	addr, val, amt := addrs[0], pks[0], int64(100)
+	sh := stake.NewHandler(sk)
+	slh := NewHandler(keeper)
+	got := sh(ctx, newTestMsgDeclareCandidacy(addr, val, amt))
+	require.True(t, got.IsOK())
+	_ = sk.Tick(ctx)
```

Comment:
just `sk.Tick(ctx)` works to (throughout all these tests)

Reply:
Updated.